object/src/mem.rs

```diff
@@ -916,6 +934,150 @@ where
         unreachable!()
     }
 
+    pub fn entry_at(
```

Comment:
I wonder if we can reuse some of the implementation from the operations API rather than repeating most of it here.

Reply:
Honestly totally forgot about that API.  As someone coming over from pydicom, I'm super used to just being able to mutate values at arbitrarily nested locations in dicoms, i.e. `dcm.PerFrameFunctionalGroups[0].ReferencedImageSequence[0].ReferencedFrameNumber = <val>`

From what I understand, I think all of the `update_value_at` functionality could be accomplished with the AttributeOp API, but (again coming from pydicom), I think having `entry_at` and `update_value_at` would be very useful shortcuts

Reply:
> I'm super used to just being able to mutate values at arbitrarily nested locations in dicoms, i.e.

As much as I am in favor of intuitive APIs (one of the aims of the project for that matter), creating one that is as intuitive as the one in pydicom would be highly unconventional in Rust. It may be feasible to write a macro which translates an expression in that format into an `AttributeOp::apply` call, although that too would require some heavy lifting.

> From what I understand, I think all of the update_value_at functionality could be accomplished with the AttributeOp API

It is possible to use one of the various actions to replicate the intended effect, although the API does not enable custom update operations, as that would severely change the nature of attribute actions. So `update_value_at` could be useful in this scenario.